solutions/cf-1528A.mdx

```diff
@@ -0,0 +1,110 @@
+---
+id: cf-1528A
+source: CF
+title: Parsa's Humongous Tree
+author: Ryan Chou
+---
+
+[Official Editorial (C++)](https://codeforces.com/blog/entry/91058)
+
+## Explanation
+
+Let's go through a case where we've already computed the beauty values for all of vertex $u$'s neighbors.
+
+Let $\texttt{range}_u$ store $l_u$ and $r_u$ and $\texttt{adj}_u$ store the beauty values of all vertices adjacent to $u$.
+
+$\texttt{range}_u = \{1, 10\}$
+
+$\texttt{adj}_u = \{2, 3, 6, 9\}$
+
+Let's say that we select $7$, can we do better?
```

Comment:
Say we select $7$ as our current vertex value, leading to a beauty value of [ASDF]. Can we do better than this?